pallets/loan/src/benchmarking.rs

```diff
@@ -0,0 +1,528 @@
+		1,
+		T::AdminOrigin::successful_origin(),
+		pool_owner.clone(),
+		Some(loan_account.clone()),
+	);
+
+	whitelist_acc::<T>(&pool_owner);
+	whitelist_acc::<T>(&loan_account);
+	(pool_owner, pool_id, loan_account, loan_class_id)
+}
+
+fn create_asset<T: Config>() -> (T::AccountId, AssetOf<T>)
+where
+	<T as pallet_balances::Config>::Balance: From<u128>,
+	<T as pallet_uniques::Config>::ClassId: From<u64>,
+{
+	// create asset
+	let loan_owner = account::<T::AccountId>("caller", 0, 0);
+	make_free_cfg_balance::<T>(loan_owner.clone());
+	let asset_class_id = create_nft_class::<T>(2, loan_owner.clone(), None);
+	let asset_instance_id = mint_nft::<T>(loan_owner.clone(), asset_class_id);
+	let asset = Asset(asset_class_id, asset_instance_id);
+	whitelist_acc::<T>(&loan_owner);
+	(loan_owner, asset)
+}
+
+fn activate_test_loan_with_defaults<T: Config>(pool_id: PoolIdOf<T>, loan_id: T::LoanId)
+where
+	<T as LoanConfig>::Rate: From<Rate>,
+	<T as LoanConfig>::Amount: From<Amount>,
+{
+	let loan_type = LoanType::BulletLoan(BulletLoan::new(
+		// advance rate 80%
+		Rate::saturating_from_rational(80, 100).into(),
+		// expected loss over asset maturity 0.15%
+		Rate::saturating_from_rational(15, 10000).into(),
+		// collateral value
+		Amount::from_inner(125 * CFG).into(),
+		// 4%
+		math::rate_per_sec(Rate::saturating_from_rational(4, 100))
+			.unwrap()
+			.into(),
+		// 2 years
+		math::seconds_per_year() * 2,
+	));
+	// interest rate is 5%
+	let rp: T::Rate = math::rate_per_sec(Rate::saturating_from_rational(5, 100))
+		.unwrap()
+		.into();
+	let origin = T::AdminOrigin::successful_origin();
+	LoanPallet::<T>::activate_loan(origin, pool_id, loan_id, rp, loan_type)
+		.expect("loan activation should not fail");
+}
+
+fn add_test_write_off_groups<T: Config>(pool_id: PoolIdOf<T>)
+where
+	<T as LoanConfig>::Rate: From<Rate>,
+{
+	for group in vec![(3, 10), (5, 15), (7, 20), (20, 30), (120, 100)] {
+		LoanPallet::<T>::add_write_off_group_to_pool(
+			T::AdminOrigin::successful_origin(),
+			pool_id,
+			WriteOffGroup {
+				percentage: Rate::saturating_from_rational(group.1, 100).into(),
+				overdue_days: group.0,
+			},
+		)
+		.expect("adding write off groups should not fail");
+	}
+}
+
+benchmarks! {
+	where_clause {
+		where
+		<T as pallet_uniques::Config>::ClassId: From<u64>,
+		<T as pallet_balances::Config>::Balance: From<u128>,
+		CurrencyIdOf<T>: From<u32>,
+		<T as LoanConfig>::Rate: From<Rate>,
+		<T as LoanConfig>::Amount: From<Amount>,
+		PoolIdOf<T>: From<<T as pallet_pool::Config>::PoolId>,
+		<T as ORMLConfig>::Balance: From<u128>,
+		<T as ORMLConfig>::CurrencyId: From<u32>,
+		<T as TimestampConfig>::Moment: From<u64> + Into<u64>,
+	}
+
+	initialise_pool {
+		let origin = T::AdminOrigin::successful_origin();
+		let pool_id: PoolIdOf<T> = Default::default();
+		let class_id: <T as LoanConfig>::ClassId = Default::default();
+		let call = Call::<T>::initialise_pool(pool_id, class_id);
+	}:{ call.dispatch_bypass_filter(origin)? }
```

Comment:
If the actual runtime has a base filter we dont know of but not necessary to test that case, we can this to bypass the filter i believe.

Reply:
I thought the benchmarks are already bypassing this, as they use `Raw::Origins`.

Reply:
I dont think so since the actual frame_system::Config impl will be actually unknown to us here, we would rather bypass it explicitly than implicitly relying on the unknown